jenkins/spark-premerge-build.sh

```diff
@@ -61,11 +61,26 @@ mvn_verify() {
     # enable UTF-8 for regular expression tests
     for version in "${SPARK_SHIM_VERSIONS_PREMERGE_UTF8[@]}"
     do
+        echo "Spark version (regex): $version"
         env -u SPARK_HOME LC_ALL="en_US.UTF-8" $MVN_CMD $MVN_URM_MIRROR -Dbuildver=$version test $MVN_BUILD_ARGS \
           -Dpytest.TEST_TAGS='' \
           -DwildcardSuites=com.nvidia.spark.rapids.ConditionalsSuite,com.nvidia.spark.rapids.RegularExpressionSuite,com.nvidia.spark.rapids.RegularExpressionTranspilerSuite
     done
 
+    # build Scala 2.13 versions
+    cd scala2.13
```

Comment:
Need we to run integration tests against scala 2.13 in the pre-merge CI?

I suggest we do it to make sure PRs can PASS the integration tests before getting merged, just as scala 2.12 does.
If not, the nightly build pipeline only run Unit Tests for all spark shims, then we can only find the integration test failures(if any) when running the nightly test CI, then it's too late.

@NVnavkumar, I am updating the CI scripts, we need to move this scala-2.13 tests into new funcion(e.g. `premerge-scala2.13`), and let it run parallely with `mvn_verify()` and `ci_2()` and will update the changes to you to help PASS the pre-merge CI.

Reply:
@NvTimLiu please also make sure the Jenkinsfile's compatibility can still cover old runs. e.g. If project version is elder than 23.12, then skip the 2.13 stage, thanks

Reply:
Updated the premerge script.